logstash-core/lib/logstash/plugins/registry.rb

```diff
@@ -0,0 +1,83 @@
+# encoding: utf-8
+require 'singleton'
+require "rubygems/package"
+
+module LogStash
+  class Registry
+
+    ##
+    # Placeholder class for registered plugins
+    ##
+    class Plugin
+      attr_reader :type, :name
+
+      def initialize(type, name)
+        @type  = type
+        @name  = name
+      end
+
+      def path
+        "logstash/#{type}s/#{name}"
+      end
+
+      def gem_name
```

Comment:
I think one of the goals of this change is to make gem names orthogonal to what plugin gets registered. I think we could make this more clear with a method name like `guessed_gem_name`. One of my goals is to be able to combine plugins like logstash-input-rabbitmq and logstash-output-rabbitmq into one gem.